### PR TITLE
Prepend space option added to bpe encoder

### PR DESCRIPTION
**Pull request by vesteinn**

### CODE REVIEW

File: src/greynirseq/utils/bpe/multiprocessing_bpe_encoder.py [hunk 1]
```diff
@@ -104,11 +104,15 @@ def initializer(self):
 
     def encode(self, line):
         global bpe
+        if self.args.add_prefix_space:
```
Comment (haukurb) [inline src/greynirseq/utils/bpe/multiprocessing_bpe_encoder.py:107]: This should check if a space is already leading.

File: src/greynirseq/utils/bpe/multiprocessing_bpe_encoder.py [hunk 2]
```diff
@@ -104,11 +104,15 @@ def initializer(self):
...
+            line = " " + line
         ids = bpe.encode(line)
         return list(map(str, ids))
 
     def decode(self, tokens):
         global bpe
+        if self.args.add_prefix_space:
```
Comment (haukurb) [inline src/greynirseq/utils/bpe/multiprocessing_bpe_encoder.py:114]: I think this should be more defensively programmed.

File: src/greynirseq/utils/bpe/multiprocessing_bpe_encoder.py [hunk 3]
```diff
@@ -50,6 +50,11 @@ def main():
         action="store_true",
         help="keep empty lines",
     )
+    parser.add_argument(
+        "--add-prefix-space",
+        action="store_true",
+        help="adds empty space at front of line before tokenizing, you probably want this"
```
Comment (haukurb) [inline src/greynirseq/utils/bpe/multiprocessing_bpe_encoder.py:56]: "If one isn't already there".

**Review by haukurb — CHANGES_REQUESTED** [review]

File: src/greynirseq/utils/bpe/multiprocessing_bpe_encoder.py (hunk 2 above)
Comment (vesteinn) [reply src/greynirseq/utils/bpe/multiprocessing_bpe_encoder.py:114]: This is the same behavior as in tokenizers, and in fashion with --keep-empty

**Pull request merged**